### DISCUSSION TITLE
✍️ Submit your project for the Showcase!

### DISCUSSION THREAD

**Discussion by djipco** [djipco/webmidi#162] [Show and tell]
The new website (currently in the works) will feature a **Showcase** section to present and promote all the amazing projects that have been created with WebMidi.js.

**If you want to add your own projet** to the showcase, simply modify [this file](https://github.com/djipco/webmidi/blob/develop/website/src/pages/showcase/index.md) and submit a pull request. Just follow the template used for the other projects (linked project name, creator and description).

Cheers!

**Comment by Pno74** [comment]
Hello djipco, 

I don't know if I'm in the right place but I'm going for it

I have been a professional developer for .... hem .... over 30 years: /

I am French, and an amateur musician (guitar and synths)

I am passionate about the Webmidi API, after buying a Kronos, I started a project for more than three years now, I work on it a few hours a day when I have my head to it.

Even this lately I haven't seen any progress on the API I still believe in it ... :)

My project is presented here:

https://kronosflyby.myweb2be.com/

And here :

https://www.youtube.com/channel/UCwR2I07meVr9z2hEiytd97Q

I will be happy to discuss and / or collaborate with the interested parties.

And I guess you must have some inside info on the future of the API?

I am rather old school, and not too used to these collaborative tools but if it is the condition to do interesting things I will do it.

I also have in mind the implementation of a daw project based on the Webmidi API ... where people could connect to the same song project etc ... :)

 I take this opportunity to find out if you are aware of the fact that once a webapp is open all the inputs of the mini devices seem inaccessible to other applications on the PC: /  

It's very annoying to do a daw :)))

I hope that google will have translated that well for me because I speak English like a Spanish cow as we say at home ....

Greetings from France :)

Patrice.

**Reply by djipco** [reply]
> I don't know if I'm in the right place but I'm going for it

That's precisely the right place for that!

I do not own a Korg wrokstation but, to owners, it seems like your software takes it to another level. 

> I take this opportunity to find out if you are aware of the fact that once a webapp is open all the inputs of the mini devices seem inaccessible to other applications on the PC: /

I have not experienced that on macOS but I'm guessing this may be related more to the system's MIDI implementation than it is to the Web MIDI API or WebMidi.js. I find it very surprising though...

Merci pour le partage Patrice!

**Comment by Pno74** [comment]
Thank you very much for your quick response.

It is true that this bug only occurs on Windows, not on Mac, but it remains very handicapping for the future of the API.

Greetings,

Patrice.

**Comment by djipco** [comment]
> It is true that this bug only occurs on Windows, not on Mac, but it remains very handicapping for the future of the API.

I was very surprised to learn about this. Unfortunately, since it happens at the system MIDI driver level, there is nothing I can do to address it in WebMidi.js. However, there is [a workaround](https://tristancalderbank.com/2020/08/19/how-to-use-the-same-midi-device-on-windows-across-multiple-programs-at-the-same-time/) that involves creating a virtual device to re-route messages.

In the upcoming version 3 of WebMidi.js, you will be able to forward MIDI messages from an input to an output (like a physical MIDI THRU port). This may also help you correctly route messages.

You can [try version 3](https://github.com/djipco/webmidi/tree/develop) (still in alpha) by going to the develop branch on GitHub and following the instructions.

**Comment by Pno74** [comment]
Hi djipco, many thanks, 

I will try the workaround, and I will also start to see to integrate WebMidi.js, but probably for the next version because there are several tens of thousands of lines to review im my case ...

Thanks again ! :)

Patrice.

**Comment by maxcomperatore** [comment]
Hi @djipco!

Submitting a project to the showcase that relies heavily on Web MIDI in the browser: [bipluk](https://bipluk.com) (Browser-native Web MIDI Hardware Synthesizer Cloud Librarian).

### What it does:
Bipluk connects directly to vintage 1980s/1990s hardware synthesizers (Yamaha DX7, Roland Juno-106, Korg M1, Sequential Prophet, Oberheim Matrix-1000) over standard class-compliant USB-to-MIDI adapters directly in Chrome:
- **Zero software or desktop drivers to install:** Runs byte-accurate bidirectional SysEx handshakes straight from the browser tab.
- **Hardware Soundbank Vaulting:** Backs up volatile battery-backed SRAM patches to the cloud before coin batteries die.
- **Real-time ASCII Name Extraction:** Unpacks raw 7-bit binary nibbles in real-time to index patch names and parameters.
- **Hardware-safe Pacing:** Automatically chunks large bulk dumps with 50-60ms intervals to respect vintage 31,250 baud CPU buffer constraints.

Huge thanks to you and the WebMidi.js community for keeping browser-based MIDI engineering vibrant!

**Reply by djipco** [reply]
That's awesome!
Can you modify the showcase file and submit a PR ?

**Comment by maxcomperatore** [comment]
Just submitted PR #619 with Bipluk added under the SysEx Librarian section! Thank you so much for the showcase feature and for keeping WebMidi.js such an awesome cornerstone for browser audio developers.

**Reply by djipco** [reply]
I just merged it. I'll push it soon to the website.

**Reply by maxcomperatore** [reply]
Awesome, thank you Jean-Philippe! Just saw it live on the showcase page. Truly appreciate you merging it and for building such an incredible foundation for the Web MIDI ecosystem.

**Reply by djipco** [reply]
My pleasure!
That's definitely a labour of love so it's always nice to receive some nice comments about it.

**Reply by maxcomperatore** [reply]
It really shows! Building on top of it has been a joy. Keep up the phenomenal work with WebMidi.js!
